### PR TITLE
[Codesmell] cleanup

### PR DESCRIPTION
**Pull request by kikkiii21**
The following are changes made to reduce the amount of code smell in the main branch
- added key prop element in array
- reduced complexity in FriendSearch.js
- updated branch with main
- reduced complexity for handleEvent in ScheduleEvent.js
- removed array index in keys for Calendar.js and CommonInputEventForm.js
- added --passWithNoTests in config file

### CODE REVIEW

**Review by EyalAzimov — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
       - node/install-packages:
           pkg-manager: npm
       - run:
-          command: npm run test
+          command: npm run --passWithNoTests
```
Comment (mnashat) [inline .circleci/config.yml:16]: Also, why is this changed?

File: src/components/FriendList/FriendSearch.js [hunk 2]
```diff
@@ -62,35 +62,63 @@ export default function FriendSearch() {
...
-                                            setFound(true);
-                                        } else {
-                                            setFound(false);
-                                        }
-
-                                    }
+                    const isInFriends = list.indexOf(res.data.email)
+                    switch (isInFriends)
+                    {
+                        case 0:
```
[the remark this answers is not in the document]
Comment (kikkiii21) [reply src/components/FriendList/FriendSearch.js:68]: isFriends outputs 0 or -1

File: .circleci/config.yml (hunk 1 above)
Comment (kikkiii21) [reply .circleci/config.yml:16]: because I removed the bogus 2+2=4 test. Since I removed it there's no more tests so I have to tell it to pass without tests othewise it will complain that its looking for tests and theres none

File: src/components/FriendList/FriendSearch.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (kikkiii21) [reply src/components/FriendList/FriendSearch.js:68]: then function indexOf returns an int

[the remark this answers is not in the document]
Comment (mnashat) [reply src/components/FriendList/FriendSearch.js:68]: `indexOf(...)` returns the index of the element you're searching for (`res.data.email` in this case), or -1 if the element isn't in the list. `isInFriends` would be 0 only if the email you're searching for is the first element in the list.

File: .circleci/config.yml (hunk 1 above)
Comment (mnashat) [reply .circleci/config.yml:16]: Ahh I see. Thanks for clarifying!

File: src/components/FriendList/FriendSearch.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (kikkiii21) [reply src/components/FriendList/FriendSearch.js:68]: the list returns only 1 element

[the remark this answers is not in the document]
Comment (kikkiii21) [reply src/components/FriendList/FriendSearch.js:68]: Screenshot 2023-03-29 at 12 34 56 PM

1- demo

2- demo1 -already a friend

3- aa -doesnt exist

4- admin2 -private profile

[the remark this answers is not in the document]
Comment (mnashat) [reply src/components/FriendList/FriendSearch.js:68]: Alright thanks for the clarification!

**Review by mnashat — APPROVED** [review]
